### PR TITLE
SQL DB Files for Driver To Init Database

### PR DESCRIPTION
**Pull request by Binary-Eater**
DB Schema files. Please review to check if correctly setup. Thanks again.

### CODE REVIEW

File: utils/db_setup/studentIsInSection.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE "studentIsInSection"(
+       id        serial,
+       start     serial,
```
Comment (JParisFerrer) [inline utils/db_setup/studentIsInSection.sql]: serials are autoincrement, these should just be int or a real date type

**Review by JParisFerrer — CHANGES_REQUESTED** [review]
all of the serials should not be serial (except for id)? since it autoincrements?

File: utils/db_setup/studentIsInSection.sql (hunk 1 above)
Comment (Binary-Eater) [reply utils/db_setup/studentIsInSection.sql]: If you do not use the default value, it acts as a normal unsigned integer type. I believe you can directly insert NULL. Postgres is wacky. Could be wrong here though but that's what I understood from reading the docs.

Comment (Binary-Eater) [reply utils/db_setup/studentIsInSection.sql]: [Postgres Numeric Datatypes](https://www.postgresql.org/docs/11/datatype-numeric.html)

Actually I probably should change it either way because

`integer` - `4 bytes` - `-2147483648 to +2147483647`
`serial` - `4 bytes` - `1 to 2147483647`

I am not sure if the documentations was incorrect due to some copy-paste magic, but it doesn't make sense for the same number of bytes for the positive range to be the same.

File: utils/db_setup/isCourseInstance.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE "isCourseInstanceOf"(
+       id       serial,
+       name     varchar(120) REFERENCES "Course"(name),
```
Comment (JParisFerrer) [inline utils/db_setup/isCourseInstance.sql]: is there a reason this is name and not something like `course_name` to match the `ci_` names? Is it just because of the name between the two `name` columns?

**Review by JParisFerrer — APPROVED** [review]
no SQL expert but the rest looks ok to me; just spot checked some tables for what I remember the ER diagram to say

**Review by PiotrGalusza — APPROVED** [review]

File: utils/db_setup/isCourseInstance.sql (hunk 2 above)
Comment (Binary-Eater) [reply utils/db_setup/isCourseInstance.sql]: Yep, I assumed when thinking of the relation `isCourseInstanceOf` I was thinking that the `name` would be synonymous with the Course's name. We can change that to `course_name` if it makes it more explicit for us.

Comment (nwalters512) [reply utils/db_setup/isCourseInstance.sql]: 100% in favor of more explicit

File: utils/db_setup/psqlInit.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+TEMP_DIR="/tmp/psql"
+
+mkdir -p ${TEMP_DIR}
+
+cp *.sql ${TEMP_DIR}
```
Comment (JParisFerrer) [inline utils/db_setup/psqlInit.sh:7]: why is this necessary?

Comment (Binary-Eater) [reply utils/db_setup/psqlInit.sh:7]: The `postgres` user doesn't have access to your home directory. Not going to tamper with people's file permissions in a script.

Comment (JParisFerrer) [reply utils/db_setup/psqlInit.sh:7]: got it, makes sense 👍

Comment (Binary-Eater) [reply utils/db_setup/psqlInit.sh:7]: If you want, I can make the script clean up the `/tmp` directory but I feel that is unnecessary.

**Review by nwalters512 — APPROVED** [review]
Looks good for now :+1: should probably rework so that Node executes this in the future but this is enough to get us prototyping quickly
